File: backend/src/main/java/com/realestatecrm/dto/LeadDtos.java
package com.realestatecrm.dto;

import java.util.UUID;

public class LeadDtos {
    public static class CreateLeadRequest {
        public String firstName;
        public String lastName;
        public String email;
        public String phone;
        public String source;
    }

    public static class LeadResponse {
        public UUID id;
        public String firstName;
        public String lastName;
        public String email;
        public String phone;
        public String source;
        public String status;
        public UUID assignedTo;
    }

    public static class StatusUpdateRequest {
        public String status;
        public String note;
    }

    public static class AssignRequest {
        public UUID assignedTo;
    }
}

File: frontend/src/dto/LeadDtos.ts
// src/dto/LeadDtos.ts

// Defines the request body for creating or updating a Lead.
export interface CreateLeadRequest {
    firstName: string;
    lastName: string;
    email: string;
    phone: string;
    source: string;
}

// Defines the full Lead object received from the API (used in LeadTable).
export interface LeadResponse {
    id: string;
    first_name: string; // Must match the snake_case from your Spring Boot backend
    last_name: string; // Must match the snake_case from your Spring Boot backend
    email?: string;
    phone?: string;
    source?: string;
    status: string;
    created_at: string;
    updated_at: string;
}